### PR TITLE
Introduce a Hyper-V backend + some fixes and docs update

### PR DESCRIPTION
**Pull request by rn**
- Add a new Hyper-V backend. The new backend provides interactive console access in powershell (and potentially bash, untested), supports networking via an existing switch and supports multiple disks
- Fix disk size code (bug introduced by #2009
- Update platform documentation
  - Use a table for the top-level README
  - Add Hyper-V documentation
  - Use a common template for some of the platforms (others should be converted)
  - Add stub docs for supported platforms not yet documented

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,14 +47,15 @@ You can use `linuxkit run <name>` or `linuxkit run <name>.<format> to execute th
...
-- [Amazon Web Services](docs/platform-aws.md)
-- [Google Cloud](docs/platform-gcp.md)
-- [Microsoft Azure](docs/platform-azure.md)
-
-We'll add more detailed docs for other platforms in the future.
+Supported platforms are:
+
+| Local                                                 | Cloud                                       |
+|-------------------------------------------------------|---------------------------------------------|
+| [HyperKit (macOS)](docs/platform-hyperkit.md)         | [Amazon Web Services](docs/platform-aws.md) |
```
Comment (justincormack) [inline README.md]: prefer two lists to a table. They arent going to continue to have the same number of items in each, and harder to read.

File: docs/platform-hyperkit.md [hunk 2]
```diff
@@ -54,7 +80,11 @@ docker run --rm -ti -v ~/.ssh:/root/.ssh  ssh ssh <IP address of VM>
 While VPNKit has the general tooling to expose any VMs port on the
 localhost (just like it does with containers in Docker for Mac), we
 are unlikely to expose this as a general feature in `linuxkit run` as
-it is very specific to the macOS. However, you can use a `socat` container to proxy between LinuxKit VMs ports and localhost.  For example, to expose the redis port from the [RedisOS example](../examples/redis-os.yml), use this Dockerfile:
+it is very specific to the macOS. However, you can use a `socat`
```
Comment (justincormack) [inline docs/platform-hyperkit.md]: This is about to be updated in https://github.com/linuxkit/linuxkit/pull/2008 to show how to do port forwarding  so maybe don't change?

File: docs/platform-hyperv.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+Example:
+```sh
+linuxkit.exe run -disk size=1 linuxkit-efi.iso
+```
+
+The Hyper-V VM, by default, is named after the prefix of the ISO, ie
+with `.iso` and `-efi` stripped. Note, You may only have one VM for a
+given name.  You can specify and alternative name using the `-name`
```
Comment (justincormack) [inline docs/platform-hyperv.md]: type and->an

File: docs/platform-hyperv.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+## Console
+
+The serial port of the VM is configured to redirect to a Named Pipe,
+and when the `linuxkit` command is executed an interactive console is
+provided in the same window. The serial console may also be redirected
+to a file.
+
+**Note:** The connection to the Named Pipe sometimes seems to be a bit
+racey, though the code it self should be fine. You may have to try
```
Comment (justincormack) [inline docs/platform-hyperv.md]: typo "it self"

File: docs/platform-hyperv.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+The Hyper-V backend supports multiple disks to be attached to the VM
+using the standard `linuxkit` `-disk` syntax. While Hyper-V typically
+stores disk images under a default location, if the VM is created with
+`linuxkit`, by default, new disks are created in the current
+directory.
+
+
+## Networking
+
+By default, the Hyper-V backend will try to find and external switch
```
Comment (justincormack) [inline docs/platform-hyperv.md]: type "and" should be "an"

File: docs/platform-qemu.md [hunk 6]
```diff
@@ -1,7 +1,33 @@
...
+
+## Console
+
+TBD: 
+
+
+## Disks
+
+The qemu backend supports multiple disks to be attached to the VM
+using the standard `linuxkit` `-disk` syntax. the qemu backend
```
Comment (justincormack) [inline docs/platform-qemu.md]: should be "The"

File: docs/platform-hyperkit.md (hunk 2 above)
Comment (rn) [inline docs/platform-hyperkit.md]: I #2008 ready to be merged? If so I handle the merge. I do would like to have the hyperkit (and other docs) follow a common pattern

File: src/cmd/linuxkit/run_hyperv.go [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+		log.Infof("Stop-VM error: %v\n%s", err, out)
+	}
+
+	log.Info("Remove the VM")
+	_, out, err = poshCmd("Remove-VM",
+		"-Name", fmt.Sprintf("'%s'", *vmName), "-Force")
+	if err != nil {
+		log.Infof("Remove-VM error: %v\n%s", err, out)
+	}
+	// TODO(rn): Should we remove disks here?
```
Comment (justincormack) [inline src/cmd/linuxkit/run_hyperv.go]: we leave them for the other local backends, so you can use them as persistent disks

**Comment by justincormack** [comment]
some minor comments but otherwise LGTM

**Comment by rn** [comment]
Addressed all the comments (thanks for the spell check)

**Pull request merged**